core/tests/bank_forks.rs

```diff
@@ -118,18 +159,27 @@ mod tests {
     // also marks each bank as root and generates snapshots
     // finally tries to restore from the last bank's snapshot and compares the restored bank to the
     // `last_slot` bank
-    fn run_bank_forks_snapshot_n<F>(last_slot: Slot, f: F, set_root_interval: u64)
-    where
+    fn run_bank_forks_snapshot_n<F>(
+        snapshot_version: SnapshotVersion,
+        last_slot: Slot,
+        f: F,
+        set_root_interval: u64,
+    ) where
         F: Fn(&mut Bank, &Keypair),
     {
         solana_logger::setup();
         // Set up snapshotting config
-        let mut snapshot_test_config = setup_snapshot_test(1);
+        let mut snapshot_test_config = SnapshotTestConfig::new(snapshot_version, 1);
 
         let bank_forks = &mut snapshot_test_config.bank_forks;
         let accounts_dir = &snapshot_test_config.accounts_dir;
         let snapshot_config = &snapshot_test_config.snapshot_config;
         let mint_keypair = &snapshot_test_config.genesis_config_info.mint_keypair;
+        let snapshot_version = bank_forks
```

Comment:
nits: Why is this reassignment of `snapshot_version` needed? We can use just `snapshot_version` in argument at here: https://github.com/solana-labs/solana/pull/10536/files#diff-60ffd8c11b11dddb54a2816436a96d70R212, or not?